### ISSUE TITLE
Corporate Proxy Authentication for Java / Maven Projects

### ISSUE THREAD

**Issue by jochenr** [redhat-developer/vscode-java#2800]
We have issues with Java/Maven based projects in VS Code due to proxy authentication in our company (using "negotiate" with fallback to BASIC).

Although we configured 

```
"http.proxy": "http://myUser:myPwd@ourProxyHost:8080",
"https.proxy": "http://myUser:myPwd@ourProxyHost:8080",
"http.proxyStrictSSL": false,
"java.jdt.ls.vmargs": "-Dhttp.proxyHost=ourProxyHost -Dhttp.proxyPort=8080 -Dhttp.proxyUser=myUser -Dhttp.proxyPassword=myPwd -Dhttps.proxyHost=ourProxyHost -Dhttps.proxyPort=8080 -Dhttps.proxyUser=myUser -Dhttps.proxyPassword=myPwd",
	
"java.configuration.maven.globalSettings": "${env:MAVEN_HOME}\\conf\\settings.xml",
"java.configuration.maven.userSettings": "${env:MAVEN_HOME}\\conf\\settings.xml",
"maven.executable.path": "${env:MAVEN_HOME}\\bin\\mvn.cmd",  
"maven.settingsFile": "${env:MAVEN_HOME}\\conf\\settings.xml",
```
in out (user) settings.json

Additionally I put our certificates to cacerts file.
Also we have the proxy configured in our Maven settings.xml (the projects build perfectly fine on commandline with maven)

The "OUTPUT" for "Language Support" for Java says:

```
WARNING: Using incubator modules: jdk.incubator.vector, jdk.incubator.foreign
[Warn  - 09:23:38] 15.11.2022, 09:23:38 Could not load Gradle version information
Cannot download published Gradle versions.
org.gradle.api.UncheckedIOException: Cannot download published Gradle versions.
	at org.eclipse.buildship.core.internal.util.gradle.PublishedGradleVersions.downloadVersionInformation(PublishedGradleVersions.java:165)
	at org.eclipse.buildship.core.internal.util.gradle.PublishedGradleVersions.tryToDownloadAndCacheVersions(PublishedGradleVersions.java:149)
	at org.eclipse.buildship.core.internal.util.gradle.PublishedGradleVersions.create(PublishedGradleVersions.java:109)
	at org.eclipse.buildship.core.internal.util.gradle.PublishedGradleVersionsWrapper$LoadVersionsJob.run(PublishedGradleVersionsWrapper.java:60)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: java.io.IOException: Unable to tunnel through proxy. Proxy returns "HTTP/1.1 407 Proxy Authentication Required"
	at java.base/sun.net.www.protocol.http.HttpURLConnection.doTunneling0(HttpURLConnection.java:2266)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.doTunneling(HttpURLConnection.java:2136)
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:185)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1665)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1589)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:224)
	at org.eclipse.buildship.core.internal.util.gradle.PublishedGradleVersions.downloadVersionInformation(PublishedGradleVersions.java:162)
	... 4 more

[Error - 09:23:43] 15.11.2022, 09:23:41 Error occured while building workspace. Details: 
 message: Project build error: Unresolveable build extension: Plugin io.quarkus.platform:quarkus-maven-plugin:2.13.3.Final or one of its dependencies could not be resolved: Failed to read artifact descriptor for io.quarkus.platform:quarkus-maven-plugin:jar:2.13.3.Final; code: 0; resource: C:/......./pom.xml; line: 1
```

 
 
I expect the problmen in the setting:
```
"java.jdt.ls.vmargs": "-Dhttp.proxyHost=ourProxyHost -Dhttp.proxyPort=8080 -Dhttp.proxyUser=myUser -Dhttp.proxyPassword=myPwd -Dhttps.proxyHost=ourProxyHost -Dhttps.proxyPort=8080 -Dhttps.proxyUser=myUser -Dhttps.proxyPassword=myPwd",
```

If I delet it, VS Code cannot connect at all (results in Timeout). So I think that User(Password from the "java.jdt.ls.vmargs" setting is not respected.

So what is the problem in my configuration?

Actually, why do I have to configure the proxy settings manually at all?
According to
ttps://code.visualstudio.com/docs/setup/network
the system/os proxy is used by VS Code (for the installation of extensions it worked, but not for usage of the java extensions)

##### Environment
- Operating System:  Windows 10 (64-bit)
- JDK version:  17 (Red Hat Build of OpenJDK "java-17-openjdk-17.0.5.0.8-2.win.x86_64")
- Visual Studio Code version: 1.73.1 (user setup)
- Java extension version: v1.12.0

**Comment by snjeza** [comment]
@jochenr could you try to set
```
"java.jdt.ls.vmargs": "-Djdk.http.auth.tunneling.disabledSchemes= -Dhttp.proxyHost=ourProxyHost -Dhttp.proxyPort=8080 -Dhttp.proxyUser=myUser -Dhttp.proxyPassword=myPwd -Dhttps.proxyHost=ourProxyHost -Dhttps.proxyPort=8080 -Dhttps.proxyUser=myUser -Dhttps.proxyPassword=myPwd",
	
```
Added `-Djdk.http.auth.tunneling.disabledSchemes=`

**Comment by jochenr** [comment]
Hello @snjeza ,

thank you very much for the hint/idea.
It helped "a little bit".
An error now happens later during importing/opening the project:

```
[Error - 08:48:28] 16.11.2022, 08:48:26 Error occured while building workspace. Details: 
 message: Project build error: 'dependencies.dependency.version' for io.quarkus:quarkus-junit5:jar is missing.; code: 0; resource: C:/Daten/git_github_jochenr/testdemo/quarkus-hello-project/pom.xml; line: 39
 message: Project build error: Non-resolvable import POM: io.quarkus.platform:quarkus-bom:pom:2.13.3.Final failed to transfer from https://repo.maven.apache.org/maven2 during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of central has elapsed or updates are forced. Original error: Could not transfer artifact io.quarkus.platform:quarkus-bom:pom:2.13.3.Final from/to central (https://repo.maven.apache.org/maven2): Connect timed out; code: 0; resource: C:/Daten/git_github_jochenr/testdemo/quarkus-hello-project/pom.xml; line: 21
```

Now it seems to be a problem with Maven, although the proxy is configured in settings.xml
On commandline (and in eclipse) maven with the same settings.xml is working.
(After importing it into eclipse I had to delete the .classpath, .project and .settings files, because the presence of seem to change the behaviour in VS Code)

I tried with my existing project and with a newly created one with the same result.

**Comment by snjeza** [comment]
@jochenr could you try to replace `${env:MAVEN_HOME}` with the absolute pathname in
```
"java.configuration.maven.globalSettings": "${env:MAVEN_HOME}\\conf\\settings.xml",
"java.configuration.maven.userSettings": "${env:MAVEN_HOME}\\conf\\settings.xml",
```

**Comment by snjeza** [comment]
@jochenr You can try to set a proxy on your Windows (Settings>Proxy), remove all proxy settings in VS Code, add `-Djdk.http.auth.tunneling.disabledSchemes=` and run VS Code.
VS Code should use the native OS settings by default.

**Comment by jochenr** [comment]
Hi @snjeza,
for the java part I'm fine now.
Thank you very much for your help.

Overall I'm almost satisfied.
This morning I did a clean new install of VS Code.
Installation and usage of the Red Hat Java and Camel extensions work quite well.
The only thing I had to use the custom proxy configuration for
```
      "http.proxyStrictSSL": false,
      "http.proxy": "http://user:password@proxy:8080",
      "https.proxy": "http://user:password@proxy:8080",
```
was the installation of the Kubernetes Extension (that gets automatically installed by the OpenShift Extension).
I can live with that. I just delete the proxy-config after installing the extensions....
